mono/mini/mini-runtime.c

```diff
@@ -3987,6 +3987,19 @@ mini_init (const char *filename, const char *runtime_version)
 #endif
 	mono_code_manager_install_callbacks (&code_manager_callbacks);
 
+	mono_profiler_state.context_enable = mini_profiler_context_enable;
+	mono_profiler_state.context_get_this = mini_profiler_context_get_this;
+	mono_profiler_state.context_get_argument = mini_profiler_context_get_argument;
+	mono_profiler_state.context_get_local = mini_profiler_context_get_local;
+	mono_profiler_state.context_get_result = mini_profiler_context_get_result;
+	mono_profiler_state.context_free_buffer = mini_profiler_context_free_buffer;
+
+	if (profile_options)
+		for (guint i = 0; i < profile_options->len; i++)
+			mono_profiler_load ((const char *) g_ptr_array_index (profile_options, i));
+
+	mono_profiler_started ();
+
 	mono_hwcap_init ();
```

Comment:
Is this the right place ? Does hwcap_init ()/arch_cpu_init () generates events ?

Reply:
`mono_arch_init` can generate events because of trampoline creation (thus raising code buffer events). I moved it before `mono_hwcap_init` also in case we decide to make that module generate events at some point.